games/applications/run_bv.py

```diff
@@ -0,0 +1,65 @@
+# ------------------------------------------------------------------------------
+# Part of Red Queen Project.  This file is distributed under the MIT License.
+# See accompanying file /LICENSE for details.
+# ------------------------------------------------------------------------------
+
+"""Benchmark Bernstein Vazirani circuits."""
+
+import pytest
+
+from qiskit import QuantumCircuit, QuantumRegister, ClassicalRegister
+
+from applications import backends, run_qiskit_circuit
+
+
+def build_bv_circuit(secret_string, mid_circuit_measure=False):
+    input_size = len(secret_string)
+    num_qubits = input_size + 1
+    if not mid_circuit_measure:
+        qr = QuantumRegister(num_qubits)
+        cr = ClassicalRegister(input_size)
+        qc = QuantumCircuit(qr, cr, name="main")
+        qc.x(qr[input_size])
+        qc.h(qr)
+        qc.barrier()
+        for i_qubit in range(input_size):
+            if secret_string[input_size - 1 - i_qubit] == "1":
+                qc.cx(qr[i_qubit], qr[input_size])
+        qc.barrier()
+        qc.h(qr)
+        qc.x(qr[input_size])
+        qc.barrier()
+        qc.measure(qr[:-1], cr)
+    else:
+        qr = QuantumRegister(2)
+        cr = ClassicalRegister(input_size)
+        qc = QuantumCircuit(qr, cr, name="main")
+        qc.x(qr[1])
+        qc.h(qr[1])
+        qc.barrier()
+        for i in range(input_size):
+            if secret_string[input_size - 1 - i] == "1":
+                qc.h(qr[0])
+                qc.cx(qr[0], qr[1])
+                qc.h(qr[0])
+            qc.measure(qr[0], cr[i])
+            qc.reset([0])
```

Comment:
Ideally you would reset q[1] at the same time and then re-init to the minus state.  This helps prevent dephasing of the target.  The QED-C people took the circuit from a research blog that I wrote that was missing this reset because Kolkata was complaining at the time.

Reply:
Done in: https://github.com/Qiskit/red-queen/pull/6/commits/c8474558c2ada26a7eee01865324354d248c5ef1